core/webapp/vis/src/utils.js

```diff
@@ -331,8 +331,8 @@ LABKEY.vis.arrayObjectIndexOf = function(myArray, searchTerm, property) {
 
 LABKEY.vis.discreteSortFn = function(a,b) {
     // Issue 23015: sort categorical x-axis alphabetically with special case for "Not in X" and "[Blank]"
-    var aIsEmptyCategory = a && (a.indexOf("Not in ") == 0 || a == '[Blank]'),
-        bIsEmptyCategory = b && (b.indexOf("Not in ") == 0 || b == '[Blank]');
+    var aIsEmptyCategory = a && a.indexOf !== undefined && (a.indexOf("Not in ") == 0 || a == '[Blank]'),
```

Comment:
would it maybe be more correct to do something like:

```suggestion
    var aIsEmptyCategory = a && typeof a === 'string' && (a.indexOf("Not in ") == 0 || a == '[Blank]'),
```

Reply:
yup, done